crates/rpc/src/server/api.rs

```diff
@@ -27,50 +25,43 @@ use miden_objects::{
     utils::serde::Deserializable,
 };
 use miden_tx::TransactionVerifier;
-use tonic::{
-    IntoRequest, Request, Response, Status, service::interceptor::InterceptedService,
-    transport::Channel,
-};
+use tonic::{IntoRequest, Request, Response, Status};
 use tracing::{debug, info, instrument};
+use url::Url;
 
 use crate::COMPONENT;
 
 // RPC SERVICE
 // ================================================================================================
 
-type StoreClient = store_client::RpcClient<InterceptedService<Channel, OtelInterceptor>>;
-type BlockProducerClient =
-    block_producer_client::ApiClient<InterceptedService<Channel, OtelInterceptor>>;
-
 pub struct RpcService {
-    store: StoreClient,
+    store: StoreRpcClient,
     block_producer: Option<BlockProducerClient>,
 }
 
 impl RpcService {
-    pub(super) fn new(
-        store_address: SocketAddr,
-        block_producer_address: Option<SocketAddr>,
-    ) -> Self {
+    pub(super) fn new(store_url: &Url, block_producer_url: Option<Url>) -> Self {
         let store = {
-            let store_url = format!("http://{store_address}");
-            // SAFETY: The store_url is always valid as it is created from a `SocketAddr`.
-            let channel = tonic::transport::Endpoint::try_from(store_url).unwrap().connect_lazy();
-            let store = store_client::RpcClient::with_interceptor(channel, OtelInterceptor);
-            info!(target: COMPONENT, store_endpoint = %store_address, "Store client initialized");
+            // SAFETY: The store_url is always valid as it is a user-provided URL that has been
+            // validated.
+            let store = Builder::new()
+                .with_address(store_url.to_string())
+                .connect_lazy::<StoreRpc>()
+                .unwrap();
```

Comment:
nit: here and in other similar places, I'd probably replace `unwrap()` with `expect()`.